File: astro/generate-pages.mjs
#!/usr/bin/env node
/**
 * Generate Astro page files for all services.
 * Reads the service config and creates minimal .astro pages.
 */
import { services } from './src/config/services.ts';
import { mkdirSync, writeFileSync } from 'fs';
import { dirname } from 'path';

for (const service of services) {
  // Skip service landing pages (pack-ship, copy-print — handled separately)
  if (['pack-ship', 'copy-print', 'home-business'].includes(service.id)) continue;

  const slug = service.slug;
  if (!slug || slug === '/' || slug === '/services') continue;

  // Determine file path from slug
  // slug looks like: /pack-ship/postage-stamps or /home-business/mailbox-rental or /copy-print/business-cards
  // or standalone: /fedex-easy-returns or /ups-store-alternative-concord-township
  const parts = slug.split('/').filter(Boolean);
  let filePath;

  if (parts.length === 1) {
    // Standalone: e.g., /fedex-easy-returns → fedex-easy-returns.astro
    filePath = `src/pages/${parts[0]}.astro`;
  } else if (parts.length >= 2) {
    // Nested: e.g., /pack-ship/postage-stamps → src/pages/pack-ship/postage-stamps.astro
    filePath = `src/pages/${parts[0]}/${parts[1]}.astro`;
  }

  if (!filePath) continue;

  // Check if this service has custom intro content (children prop in the React version)
  // We can't easily detect this from config, so create a basic template
  mkdirSync(dirname(filePath), { recursive: true });

  const content = `---
import ServiceLayout from '../../layouts/ServiceLayout.astro';
import { getServiceById } from '../../utils/services-helpers';

const service = getServiceById('${service.id}')!;
---

<ServiceLayout service={service} />
`;

  writeFileSync(filePath, content);
}

File: astro/generate_pages.py
#!/usr/bin/env python3
"""Generate Astro page files for all services."""

import os
import re
import glob

BASE = '/home/blackghost/mailbox-plus-website/astro'

# Extract id -> slug mappings from all service config files
id_to_slug = {}
skip_ids = {'pack-ship', 'copy-print', 'home-business'}

for f in (sorted(glob.glob(f'{BASE}/src/config/services/**/*.ts', recursive=True)) +
          sorted(glob.glob(f'{BASE}/src/config/micro-problems/**/*.ts', recursive=True))):
    with open(f) as fh:
        content = fh.read()
    # Extract all id/slug pairs — they appear in order in each array
    ids = re.findall(r"id: '([^']+)'", content)
    slugs = re.findall(r"slug: '([^']+)'", content)
    for i, s in zip(ids, slugs):
        if i not in skip_ids:
            id_to_slug[i] = s

print(f"Found {len(id_to_slug)} services with slugs")

created = 0
skipped = 0
for sid, slug in sorted(id_to_slug.items()):
    parts = slug.strip('/').split('/')
    
    if len(parts) == 1:
        file_path = f'{BASE}/src/pages/{parts[0]}.astro'
        layout_path = '../layouts/ServiceLayout.astro'
        utils_path = '../utils/services-helpers'
    elif len(parts) >= 2:
        os.makedirs(f'{BASE}/src/pages/{parts[0]}', exist_ok=True)
        file_path = f'{BASE}/src/pages/{parts[0]}/{parts[1]}.astro'
        layout_path = '../../layouts/ServiceLayout.astro'
        utils_path = '../../utils/services-helpers'
    else:
        continue
    
    if os.path.exists(file_path):
        skipped += 1
        continue
    
    content = f"""---
import ServiceLayout from '{layout_path}';
import {{ getServiceById }} from '{utils_path}';

const service = getServiceById('{sid}')!;
---

<ServiceLayout service={{service}} />
"""
    with open(file_path, 'w') as f:
        f.write(content.lstrip())
    created += 1
    rel_path = os.path.relpath(file_path, BASE)
    print(f"  CREATED: {rel_path}  ({slug})")

print(f"\nDone: {created} created, {skipped} skipped (already exist)")
